### DISCUSSION TITLE
Is there is way to make Type.Date into Type.String when converting the DateTime from Prisma to TypeBox?

### DISCUSSION THREAD

**Discussion by brucelin006** [m1212e/prismabox#10] [Ideas]
I am using Fastify, and I am trying to use the generated TypeBox type as the schema for the request and reply, however Fastify can only serialize the `Type.Date` if it is `Type.String`. Is there a way that Prismabox can have an option of choosing either `Type.Date` or `Type.String`?

Also, I am experiencing import issues for ESM, where the `__nullable__` generated in the prismabox directory gives an error saying it has to be `import { __nullable__ } from "./__nullable__.js";`, but the original one does not come with `.js`, is there a way to append this by Prismabox?

**Comment by m1212e** [comment]
> Also, I am experiencing import issues for ESM, where the __nullable__ generated in the prismabox directory gives an error saying it has to be import { __nullable__ } from "./__nullable__.js";, but the original one does not come with .js, is there a way to append this by Prismabox?

This seems like a misconfigured typescript/module resolution. When using ESM you should not be required to set the .js extension in your imports.

> I am using Fastify, and I am trying to use the generated TypeBox type as the schema for the request and reply, however Fastify can only serialize the Type.Date if it is Type.String. Is there a way that Prismabox can have an option of choosing either Type.Date or Type.String?

This would cause type mismatches when using the schemas with prisma. You then could not pass them into or apply them to values coming/going in/out of prisma and are required to make manual mappers. This is because prisma enforces usage of JS Date objects.
I see this more as a fastify/serialization issue since prismabox tries to stick as close to the prisma specifications as possible. I'd recommend checking out serialization options or middlewares in fastify to tackle this issue.

**Reply by m1212e** [reply]
Sidenote: I can hightly recommend [elysia](https://elysiajs.com/) over fastify, if you have a choice in tooling!

**Reply by m1212e** [reply]
Regarding the string thing you mentioned: Is [this](https://github.com/elysiajs/eden/issues/104) maybe related?

**Comment by m1212e** [comment]
This might be resolved by https://github.com/m1212e/prismabox/pull/11
